tornado/routing.py

```diff
@@ -16,73 +16,79 @@
 
 Tornado routes HTTP requests to appropriate handlers using `Router` class implementations.
 
-Any `Router` implementation can be used directly as a ``request_callback`` for
-`~.httpserver.HTTPServer` (this is possible because `Router` implements
-`~.httputil.HTTPServerConnectionDelegate`):
+`Router` interface extends `~.httputil.HTTPServerConnectionDelegate` to provide additional
+routing capabilities. This also means that any `Router` implementation can be used directly
+as a ``request_callback`` for `~.httpserver.HTTPServer` constructor.
+
+`Router` subclass must implement a ``find_handler`` method to provide a suitable
+`~.httputil.HTTPMessageDelegate` instance to handle the request:
 
 .. code-block:: python
 
     class CustomRouter(Router):
         def find_handler(self, request, **kwargs):
-            # some routing logic providing a suitable HTTPMessageDelegate implementation
-            return HTTPMessageDelegateImplementation()
+            # some routing logic providing a suitable HTTPMessageDelegate instance
+            return MessageDelegate(request.connection)
```

Comment:
I've added this example to docs and tests: https://github.com/tornadoweb/tornado/pull/1916/commits/65e4835fd44c5e4042505c924de2a25caba98e0e